packages/frontend/src/redux/slices/staking/index.js

```diff
@@ -1,4 +1,52 @@
+import { createSelector } from "reselect";
+
+import createParameterSelector from "../createParameterSelector";
+
 const SLICE_NAME = 'staking';
 
+const getAccountIdParam = createParameterSelector((params) => params.accountId);
+const getValidatorIdParam = createParameterSelector((params) => params.validatorId);
+
 // Top level selectors
-export const selectStakingSlice = (state) => state[SLICE_NAME];
+export const selectStakingSlice = (state) => state[SLICE_NAME] || {};
+
+// accounts - state
+export const selectStakingAccounts = createSelector(selectStakingSlice, (staking) => staking.accounts || []);
+
+export const selectStakingCurrentAccountbyAccountId = createSelector(
+    [selectStakingAccounts, getAccountIdParam], 
+    (accounts, accountId) => accounts.find((account) => account.accountId === accountId)
+);
+
+export const selectStakingAccountsFirst = createSelector(selectStakingAccounts, (accounts) => accounts[0] || {});
+
+export const selectStakingAccountsSecond = createSelector(selectStakingAccounts, (accounts) => accounts[1] || {});
+
+// lockup - state
+export const selectStakingLockup = createSelector(selectStakingSlice, (staking) => staking.lockup || {});
+
+export const selectStakingLockupId = createSelector(selectStakingLockup, (lockup) => lockup.lockupId || '');
+
+export const selectStakingContract = createSelector(selectStakingLockup, (lockup) => lockup.contract || {});
+
+// allValidators - state
+export const selectStakingAllValidators = createSelector(selectStakingSlice, (staking) => staking.allValidators || []);
+
+export const selectStakingAllValidatorsLength = createSelector(selectStakingAllValidators, (allValidators) => allValidators.length);
+
+export const selectStakingFindContractByValidatorId = createSelector(
+    [selectStakingAllValidators, getValidatorIdParam],
+    (allValidators, validatorId) => allValidators.find((validator) => validator.accountId === validatorId)?.contract || {}
+);
+
+// accountsObj - state
+export const selectStakingAccountsObj = createSelector(selectStakingSlice, (staking) => staking.accountsObj || {});
```

Comment:
I didn't notice any references to this selector outside this module, I think it's worth considering not exporting this if you don't foresee any code wanting to get the whole object back in the future.

Reply:
Done.